### PR TITLE
Add ability to hide progress bars

### PR DESCRIPTION
**Pull request by mattcieslak**
This PR adds an attribute to the model base class that controls progress bars in the subclasses.

### CODE REVIEW

**Comment by daducci** [comment]
Hi Matt,

I took inspiration from your PR and implemented a more general way to control what is printed during the execution of AMICO (message, errors, progress bars etc). This is now controlled via the function `amico.set_verbose( verbose )`, where the parameter `verbose` has the following meaning:
> 3 = show everything
> 2 = show messages but no progress bars
> 1 = show only warnings/errors
> 0 = hide everything

To achieve the funcitonality that you wanted, i.e. remove the progress bars, you could do something like this:
```python
import amico
amico.set_verbose( 2 )
```

Is this what you wanted? If so, could you give it a shot?

**Comment by mattcieslak** [comment]
Definitely, thanks!

**Comment by mattcieslak** [comment]
can confirm this works perfectly

**Comment by daducci** [comment]
Hola @MarioOcampo, could you try on your machine? Besides adding the possibility to control the verbosity, I also replaced `python-spams` with `spams` (which is supposed to fix some installation problems, as discussed some weeks ago)

File: setup.py [hunk 1]
```diff
@@ -14,6 +14,6 @@
       url=info.URL,
       license=info.LICENSE,
       packages=find_packages(),
-      setup_requires=['numpy>=1.12,<1.22'],
```
Comment (MarioOcampo) [inline setup.py:17]: Same change is required in requirements.txt

**Review by MarioOcampo — CHANGES_REQUESTED** [review]
I tested the PR with the getting started tutorial, it works well. 
The installation of `spams` in a new environment (python 3.7 and python 3.8) works well.  
Only the file requirements.txt needs to be updated 
Good job!!

File: setup.py (hunk 1 above)
Comment (daducci) [reply setup.py:17]: Is it ok now?

Comment (MarioOcampo) [reply setup.py:17]: Yes

**Review by MarioOcampo — APPROVED** [review]

**Pull request merged**